omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_4/EncounterSearchHandler2_4.java

```diff
@@ -0,0 +1,114 @@
+                    .withRequiredParameters("patient").withOptionalParameters("encounterType", DATE_FROM, DATE_TO, ENCOUNTER_FORMS, OBS_CONCEPTS, "order")
+                    .build()));
+
+    @Override
+    public SearchConfig getSearchConfig() {
+        return this.searchConfig;
+    }
+
+    @Override
+    public PageableResult search(RequestContext context) throws ResponseException {
+        String patientUuid = context.getRequest().getParameter("patient");
+        String encounterTypeUuid = context.getRequest().getParameter("encounterType");
+
+        String dateFrom = context.getRequest().getParameter(DATE_FROM);
+        String dateTo = context.getRequest().getParameter(DATE_TO);
+
+        String forms = context.getRequest().getParameter(ENCOUNTER_FORMS);
+        String concepts = context.getRequest().getParameter(OBS_CONCEPTS);
+
+        Date fromDate = dateFrom != null ? (Date) ConversionUtil.convert(dateFrom, Date.class) : null;
+        Date toDate = dateTo != null ? (Date) ConversionUtil.convert(dateTo, Date.class) : null;
+
+        Patient patient = ((PatientResource1_8) Context.getService(RestService.class).getResourceBySupportedClass(
+                Patient.class)).getByUniqueId(patientUuid);
+        EncounterType encounterType = ((EncounterTypeResource1_8) Context.getService(RestService.class)
+                .getResourceBySupportedClass(EncounterType.class)).getByUniqueId(encounterTypeUuid);
+
+        List<Form> formList = new ArrayList<>();
+        if (!Strings.isNullOrEmpty(forms)) {
+            FormService formService = Context.getFormService();
+            Arrays.stream(forms.split(",")).map(formService::getFormByUuid).filter(Objects::nonNull).forEach(formList::add);
+        }
+
+        List<String> conceptUuidList = Strings.isNullOrEmpty(concepts) ? new ArrayList<>() : Arrays.asList(concepts.split(","));
+
+        if (patient != null) {
+            EncounterSearchCriteriaBuilder encounterSearchCriteriaBuilder = new EncounterSearchCriteriaBuilder()
+                    .setPatient(patient).setFromDate(fromDate).setToDate(toDate).setIncludeVoided(false);
+            if (encounterType != null) {
+                encounterSearchCriteriaBuilder.setEncounterTypes(Collections.singletonList(encounterType));
+            }
+            encounterSearchCriteriaBuilder.setEnteredViaForms(formList);
+
+            EncounterSearchCriteria encounterSearchCriteria = encounterSearchCriteriaBuilder.createEncounterSearchCriteria();
+
+            List<Encounter> encounters = Context.getEncounterService().getEncounters(encounterSearchCriteria);
+            encounters.forEach(encounter -> {
+                if (!conceptUuidList.isEmpty()) {
+                    List<Obs> obs = encounter.getObs().stream().filter(ob -> conceptUuidList.contains(ob.getConcept().getUuid())).collect(Collectors.toList());
+                    encounter.setObs(new HashSet<>(obs));
```

Comment:
Don't modify the encounter object directly. This is very dangerous since it could result in an actual update to the database.